src/sentry/sentry_metrics/consumers/indexer/processing.py

```diff
@@ -53,15 +61,18 @@ def process_messages(
     else:
         # This, instead of importing the normal way, was likely done to prevent
         # fork-safety issues.
-        indexer = get_indexer()
+        if config.db_backend == CLOUDSPANNER_DB:
+            indexer = get_spanner_indexer()
+        else:
+            indexer = get_indexer()
```

Comment:
```suggestion
            indexer = get_postgres_indexer()
```

Reply:
You should also change the import to only import the postgres one. Right now it may also be loading the cloudspanner one depending on your settings.